drivers/azure/util.go

```diff
@@ -164,6 +164,32 @@ func (d *Driver) ipAddress(ctx context.Context) (ip string, err error) {
 	return ip, nil
 }
 
+// resolveNSGReference extracts the NetworkSecurityGroupID from nsg
+func (d *Driver) resolveNSGReference(nsg string) (azure.Resource, error) {
+	if strings.Contains(nsg, "/") {
```

Comment:
This places the restriction that a user can not use a `/` in their name or the parse will always fail.